conanfile.py

```diff
@@ -17,38 +18,75 @@ class WinflexbisonConan(ConanFile):
-    _source_subfolder = "source_subfolder"
+    settings = "arch", "os_build", "arch_build", "compiler", "build_type"
 
-    def config_options(self):
-        if self.settings.os != "Windows":
-            raise ConanInvalidConfiguration("winflexbison is only supported on Windows.")
+    _source_subfolder = "sources"
+
+    def configure(self):
+        self.settings.arch = str(self.settings.arch_build)
+        if self.settings.compiler != "Visual Studio":
+            raise ConanInvalidConfiguration("winflexbison is only supported for Visual Studio.")
+
+    def package_id(self):
+        self.info.include_build_settings()
+        del self.info.settings.arch
+        del self.info.settings.compiler
+        del self.info.settings.build_type
 
     def source(self):
-        tools.get("{0}/archive/v{1}.tar.gz".format(self.homepage, self.version), sha256="39fe57de6a52dc83c8a9ece90b8484d8d2b764e24e22e30ba5dc018328019a4d")
-        extracted_dir = self.name + "-" + self.version
+        name = "winflexbison"
+        url = "https://github.com/lexxmark/winflexbison/archive/v{}.tar.gz".format(self.version)
+        sha256 = "2ab4c895f9baf03dfdfbb2dc4abe60e87bf46efe12ed1218c38fd7761f0f58fc"
+        filename = "{}-{}.tar.gz".format(name, self.version)
+
+        dlfilepath = os.path.join(tempfile.gettempdir(), filename)
+        if os.path.exists(dlfilepath) and not tools.get_env("WINFLEXBISON_FORCE_DOWNLOAD", False):
+            self.output.info("Skipping download. Using cached {}".format(dlfilepath))
+        else:
+            self.output.info("Downloading {} from {}".format(name, url))
+            tools.download(url, dlfilepath)
+        tools.check_sha256(dlfilepath, sha256)
+        tools.untargz(dlfilepath)
+
+        extracted_dir = "{}-{}".format(name, self.version)
         os.rename(extracted_dir, self._source_subfolder)
+
         # Generate license from header of a source file
-        with open("%s/%s/bison/data/glr.cc" % (self.source_folder, self._source_subfolder)) as f:
+        with open(os.path.join(self.source_folder, self._source_subfolder, "bison", "data", "skeletons", "glr.cc"), ) as f:
             content_lines = f.readlines()
         license_content = []
         for i in range(2, 16):
             license_content.append(content_lines[i][2:-1])
-        tools.save("%s/%s/LICENSE" % (self.source_folder, self._source_subfolder), "\n".join(license_content))
+        tools.save(os.path.join(self.source_folder, self._source_subfolder, "COPYING"), "\n".join(license_content))
+
+    @property
+    def _fake_package(self):
+        return os.path.join(self.build_folder, "package")
 
     def build(self):
         cmake = CMake(self)
+        cmake.definitions["CMAKE_INSTALL_PREFIX"] = self._fake_package
         cmake.configure()
-        cmake.build()
+        cmake.build(args=["--config", str(self.settings.build_type)])
 
     def package(self):
-        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
```

Comment:
What is about the LICENSE file for this recipe? It should be copied as well

Reply:
It is called `COPYING`

Reply:
Maybe for winflexbison itself, but not our recipe -> https://github.com/bincrafters/conan-winflexbison/commit/58da2e0afee11823199fec7d3fbee12e4cfabc09#diff-37854d19817c792316d481f5beb93cc7

All licenses of the packaged software + the license for the Conan recipe should be packaged

Reply:
Indeed, it is not copied.
In my defence, this was also not done before. See [manifest](https://dl.bintray.com/bincrafters/public-conan/bincrafters/winflexbison/2.5.16/stable/package/3ca85915ce57a6607676849c47c2cee02c75209c/conanmanifest.txt)
I will add this.

Reply:
I've added it.